lang/java/avro/src/main/java/org/apache/avro/generic/GenericData.java

```diff
@@ -184,11 +195,11 @@ public <T> Conversion<T> getConversionByClass(Class<T> datumClass, LogicalType l
    * @return the conversion for the logical type, or null
    */
   @SuppressWarnings("unchecked")
-  public Conversion<Object> getConversionFor(LogicalType logicalType) {
```

Comment:
Do you have any idea if changing this signature is backwards binary-compatible ?  If not, we might need to bump these lines of code to a major release.

Reply:
Yes, they are:
1. The type erasure of a type variable is the erasure of its leftmost bound (https://docs.oracle.com/javase/specs/jls/se11/html/jls-4.html#jls-4.6)
2. Because T has no bounds in the syntax, it's bound is `Object` (https://docs.oracle.com/javase/specs/jls/se11/html/jls-4.html#jls-4.4)

So as far as binary compatibility is concerned, `<T> Conversion<T>` is the same as `Conversion<Object>`.